### ISSUE TITLE
v0.1.17 install fails with node 10 on Linux

### ISSUE THREAD

**Issue by PeachScript** [abbr/deasync#124]
## Versions

OS: CentOS Linux release 7.6.1810 (Core)
Node.js: 10
deasync: 0.1.17
npm: 6

## What happens

I get the`script.install` execute error when install deasync@0.1.17 with npm, error log looks like this:

``` bash
scripts.install some-dependency@version › deasync-promise@1.0.1 › deasync@^0.1.7 run "node ./build.js", root: "/path/to/project/node_modules/_deasync@0.1.17@deasync"
`linux-x64-node-10` exists; testing
Problem with the binary; manual build incoming
stdout=
err=Error: Command failed: /path/to/project/node_modules/node/bin/node quick-test.js
/path/to/project/node_modules/_bindings@1.5.0@bindings/bindings.js:135
  throw err;
  ^

Error: Could not locate the bindings file. Tried:
 → /path/to/project/node_modules/_deasync@0.1.17@deasync/build/deasync.node
 → /path/to/project/node_modules/_deasync@0.1.17@deasync/build/Debug/deasync.node
 → /path/to/project/node_modules/_deasync@0.1.17@deasync/build/Release/deasync.node
 → /path/to/project/node_modules/_deasync@0.1.17@deasync/out/Debug/deasync.node
 → /path/to/project/node_modules/_deasync@0.1.17@deasync/Debug/deasync.node
 → /path/to/project/node_modules/_deasync@0.1.17@deasync/out/Release/deasync.node
 → /path/to/project/node_modules/_deasync@0.1.17@deasync/Release/deasync.node
 → /path/to/project/node_modules/_deasync@0.1.17@deasync/build/default/deasync.node
 → /path/to/project/node_modules/_deasync@0.1.17@deasync/compiled/10.16.3/linux/x64/deasync.node
 → /path/to/project/node_modules/_deasync@0.1.17@deasync/addon-build/release/install-root/deasync.node
 → /path/to/project/node_modules/_deasync@0.1.17@deasync/addon-build/debug/install-root/deasync.node
 → /path/to/project/node_modules/_deasync@0.1.17@deasync/addon-build/default/install-root/deasync.node
 → /path/to/project/node_modules/_deasync@0.1.17@deasync/lib/binding/node-v64-linux-x64/deasync.node
     at bindings (/path/to/project/node_modules/_bindings@1.5.0@bindings/bindings.js:126:9)
     at Object.<anonymous> (/path/to/project/node_modules/_deasync@0.1.17@deasync/index.js:30:31)
     at Module._compile (internal/modules/cjs/loader.js:778:30)
     at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
     at Module.load (internal/modules/cjs/loader.js:653:32)
     at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
     at Function.Module._load (internal/modules/cjs/loader.js:585:3)
     at Module.require (internal/modules/cjs/loader.js:692:17)
     at require (internal/modules/cjs/helpers.js:25:18)
     at Object.<anonymous> (/path/to/project/node_modules/_deasync@0.1.17@deasync/quick-test.js:3:13)
 
 events.js:174
       throw er; // Unhandled 'error' event
       ^
 
 Error: spawn node-gyp ENOENT
     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:19)
     at onErrorNT (internal/child_process.js:415:16)
     at process._tickCallback (internal/process/next_tick.js:63:19)
 Emitted 'error' event at:
     at Process.ChildProcess._handle.onexit (internal/child_process.js:246:12)
     at onErrorNT (internal/child_process.js:415:16)
     at process._tickCallback (internal/process/next_tick.js:63:19)
```

But instlal deasync@0.1.16 on Linux is fine, and install deasync@0.1.17 on macOS is also fine, is the latest version contains any breaking change for Linux?

Thanks for your awesome project ❤️

**Comment by abbr** [comment]
Cannot reproduce. What if you run `npm i deasync` from an empty folder?

**Comment by PeachScript** [comment]
No lucky, I got this error when I compile a docker image for a real project on a compile server, for reproduce, I installed deasync again in a empty folder on a normal cloud server, got the same error.

**Comment by PeachScript** [comment]
The additional error log maybe help to find the reason:

``` bash
make: entering directory “/path/to/project/node_modules/deasync/build”
  CXX(target) Release/obj.target/deasync/src/deasync.o
make: g++：command not found
make: *** [Release/obj.target/deasync/src/deasync.o] error 127
make: leaving directory “/path/to/project/node_modules/deasync/build”
```
If I change the node version via `nvm` on the same server, it works fine.

**Comment by abbr** [comment]
For unknown reason pre-compiled binary on Ubuntu doesn't work on Centos but the other way around works. Just published deasync@0.1.18 to replace the binary for node 10.

**Comment by PeachScript** [comment]
Nice work! It works prefect now, thanks a lot ❤️
